### ISSUE TITLE
Plugin triggers fatal error in wordpress (TypeError in wp-includes/blocks.php:20)

### ISSUE THREAD

**Issue by sjn** [Automattic/wordpress-activitypub#482]
### Quick summary

Wordpress returns the "White page" error when activitypub plugin is activated.

The following shows up in the error.log file:

```
[Mon Oct 02 10:19:36.683485 2023] [php:error] [pid 3968784] [client 84.215.58.44:41612] PHP Fatal error:  Uncaught TypeError: strpos(): Argument #1 ($haystack) must be of type string, array given in /usr/share/wordpress/wp-includes/blocks.php:20
```

Stack trace included below.

I'm running this on Ubuntu's 22.04.3 `wordpress` package, version `5.8.3+dfsg1-1ubuntu1.1`.
This was tested with both version 1.0.0 and 1.0.1 of the activitypub plugin.

Disabling the activitypub plugin makes the wordpress instance function as expected.

### Steps to reproduce

1. On Ubuntu's 22.04.3 wordpress package, version 5.8.3+dfsg1-1ubuntu1.1, enable the activitypub plugin.

### What you expected to happen

Wordpress site shows normally

### What actually happened

"White page" error ("something went wrong, see the troubleshooting FAQ") shows. No other content.

### Impact

One

### Available workarounds?

No and the platform is unusable

### Logs or notes

```
[Mon Oct 02 10:19:36.683485 2023] [php:error] [pid 3968784] [client 84.215.58.44:41612] PHP Fatal error:  Uncaught TypeError: strpos(): Argument #1 ($haystack) must be of type string, array given in /usr/share/wordpress/wp-includes/blocks.php:20
Stack trace:
#0 /usr/share/wordpress/wp-includes/blocks.php(20): strpos()
#1 /usr/share/wordpress/wp-includes/blocks.php(142): remove_block_asset_path_prefix()
#2 /usr/share/wordpress/wp-includesblocks.php(317): register_block_style_handle()
#3 /var/lib/wordpress/wp-content/plugins/activitypub/includes/class-blocks.php(35): register_block_type_from_metadata()
#4 /var/lib/wordpress/wp-content/plugins/activitypub/includes/class-blocks.php(10): Activitypub\\Blocks::register_blocks()
#5 /var/lib/wordpress/wp-content/plugins/activitypub/activitypub.php(62): Activitypub\\Blocks::init()
#6 /usr/share/wordpress/wp-includes/class-wp-hook.php(303): Activitypub\\init()
#7 /usr/share/wordpress/wp-includes/class-wp-hook.php(327): WP_Hook->apply_filters()
#8 /usr/share/wordpress/wp-includes/plugin.php(470): WP_Hook->do_action()
#9 /usr/share/wordpress/wp-settings.php(578): do_action()
#10 /usr/share/wordpress/wp-config.php(63): require_once('...')
#11 /usr/share/wordpress/wp-load.php(50): require_once('...')
#12 /usr/share/wordpress/wp-blog-header.php(13): require_once('...')
#13 /usr/share/wordpress/index.php(17): require('...')
#14 {main}
  thrown in /usr/share/wordpress/wp-includes/blocks.php on line 20, referer: https://mysite.example/wp-admin/plugins.php
```

**Comment by mattwiebe** [comment]
Thanks for the detailed bug report and sorry for the trouble!

Do you have any other plugins active?

**Comment by sjn** [comment]
Yes, I have 4 other plugins active, and they are updated and function as expected.

**Comment by mattwiebe** [comment]
Thanks @sjn - unfortunately you are running a pretty old version of WordPress there that doesn't fully support the way we're using the WP Blocks API. In my own testing WP 5.9 already resolves the issue but as always in WordPress, running the latest release is recommended. I'll update the plugin to prevent activation on < 5.9.

**Comment by sjn** [comment]
Oof. I'm running what Ubuntu LTS offers, and don't have the capacity (or inclination) to keep it updated manually. 😟

**Comment by pfefferle** [comment]
hmmm... what do you think about https://github.com/Automattic/wordpress-activitypub/pull/484 instead @mattwiebe ???

**Comment by sjn** [comment]
BTW, the [activitypub plugin page](https://wordpress.org/plugins/activitypub/#developers) claims it supports wordpress version 4.7 or higher, so it seems there's some dependency metadata that needs updating too.

With that said, if it's possible to have this plugin support wordpress versions that are running on LTS installations out there, that would be great! I'd hate to have to choose between OS stability/predictability and having my blog connected with the fediverse.

**Comment by mattwiebe** [comment]
> hmmm... what do you think about #484 instead @mattwiebe ???

That seems like a very reasonable approach to me, since we already have the `site_supports_blocks` abstraction working for us.

**Comment by sjn** [comment]
Hei!

I just upgraded to 1.0.3, and got an error when I attempt activating the plugin:

```
Fatal error: Uncaught Error: Undefined constant "Activitypub\ACTIVITYPUB_DISABLE_REWRITES" in /var/lib/wordpress/wp-content/plugins/activitypub/includes/class-activitypub.php:232
Stack trace:
#0 /var/lib/wordpress/wp-content/plugins/activitypub/includes/class-activitypub.php(270): Activitypub\Activitypub::add_rewrite_rules()
#1 /var/lib/wordpress/wp-content/plugins/activitypub/includes/class-activitypub.php(45): Activitypub\Activitypub::flush_rewrite_rules()
#2 /usr/share/wordpress/wp-includes/class-wp-hook.php(303): Activitypub\Activitypub::activate()
#3 /usr/share/wordpress/wp-includes/class-wp-hook.php(327): WP_Hook->apply_filters()
#4 /usr/share/wordpress/wp-includes/plugin.php(470): WP_Hook->do_action()
#5 /usr/share/wordpress/wp-admin/plugins.php(193): do_action()
#6 {main} thrown in /var/lib/wordpress/wp-content/plugins/activitypub/includes/class-activitypub.php on line 232
```

I'm using the same version of PHP (8.1.2-1ubuntu2.14) and Wordpress (5.8.3+dfsg1-1ubuntu1.1) as earlier.

**Comment by pfefferle** [comment]
fixed in 1.0.4 thanks a lot for reporting @sjn

**Comment by sjn** [comment]
Upgraded to 1.0.4. It works great now. Thanks! :-D
